source/agent/addons/quic/QuicTransportStream.cc

```diff
@@ -202,20 +205,30 @@ void QuicTransportStream::SignalOnData()
 
     while (m_stream->ReadableBytes() > 0) {
         auto readableBytes = m_stream->ReadableBytes();
-        uint8_t* buffer = new uint8_t[readableBytes];
-        if (buffer == nullptr) {
-            ELOG_ERROR("Failed to alloc buffer.");
-            return;
+        if (readableBytes > m_bufferSize) {
+            ReallocateBuffer(readableBytes);
         }
         owt_base::Frame frame;
         frame.format = owt_base::FRAME_FORMAT_DATA;
         frame.length = readableBytes;
-        frame.payload = buffer;
+        frame.payload = m_buffer;
         m_stream->Read(frame.payload, readableBytes);
         deliverFrame(frame);
     }
 }
 
+void QuicTransportStream::ReallocateBuffer(size_t size)
+{
+    if (size > m_bufferSize) {
+        delete[] m_buffer;
+    }
+    m_buffer = new uint8_t[size];
+    m_bufferSize = size;
+    if (m_buffer) {
```

Comment:
if !m_buffer.  
Also return bool so that caller knows the buffer is invalid.

Reply:
What can we do if allocation failed?

Reply:
Fixed m_buffer.